GVFS/GVFS.Upgrader/UpgradeOrchestrator.cs

```diff
@@ -164,7 +169,7 @@ private bool TryRunUpgrade(out Version newVersion, out string consoleError)
 
             if (!this.upgrader.UpgradeAllowed(out error))
             {
-                ProductUpgrader.DeleteAllInstallerDownloads();
+                this.upgrader.DeleteAllInstallerDownloads(this.tracer);

```

Comment:
@jamill @alameenshah should we actually always be calling `DeleteAllInstallerDownloads` in this method?

I believe the idea is that the upgrader is always supposed to re-download the upgrades, but it wasn't clear to me what the NuGet and GitHub upgraders would do if the package that we want is already on disk in the downloads folder.

Reply:
I think this is OK - if this check fails, it is most likely due to the fact that GVFS is not configured for upgrades. If something was downloaded already, then the most likely scenario is that someone changed the config after a package was downloaded...

@alameenshah - what do you think?

related - the proposed change in #736 no longer downloads the update in the background

Reply:
> I think this is OK - if this check fails, it is most likely due to the fact that GVFS is not configured for upgrades. If something was downloaded already, then the most likely scenario is that someone changed the config after a package was downloaded...

I think I could have worded my initial comment a bit better.  To clarify I'm wondering if `DeleteAllInstallerDownloads()` should be moved out of the check and always called.

If a file already exists at the download location, will the NuGet and GitHub upgraders overwrite it?  Based on the offline conversations we've had I thought that we wanted to make sure that the installer would always run a freshly downloaded copy.

Reply:
Per conversation with @jamill offline, the upgraders will delete any existing files if present